ios/FluentUI/Bottom Commanding/BottomCommandingController.swift

```diff
@@ -534,9 +525,39 @@ open class BottomCommandingController: UIViewController {
         }
     }
 
-    private func updateSheetExpandedContentHeight() {
-        if let bottomSheetController = bottomSheetController {
-            bottomSheetController.preferredExpandedContentHeight = estimatedTableViewHeight + Constants.BottomSheet.expandedContentTopMargin
+    private func updateSheetSizingParameters() {
+        guard let bottomSheetController = bottomSheetController else {
+            return
+        }
+        bottomSheetController.isExpandable = isExpandable
+
+        let maxHeroItemHeight = heroCommandStack.arrangedSubviews.map { $0.intrinsicContentSize.height }.max() ?? Constants.defaultHeroButtonHeight
+
+        if isExpandable {
+            let hasTallHeroButton = maxHeroItemHeight > Constants.defaultHeroButtonHeight
+
+            // Sizing that depends on how tall the current hero commands are
+            let expandedContentTopMargin = hasTallHeroButton
+                ? Constants.BottomSheet.expandedContentReducedTopMargin
+                : Constants.BottomSheet.expandedContentRegularTopMargin
+            expandedContentTopConstraint?.constant = expandedContentTopMargin
+
+            // How tall should the expanded content be
+            bottomSheetController.preferredExpandedContentHeight = estimatedTableViewHeight + expandedContentTopMargin
+        }
+
+        // Header top margin differs depending on expandability
+        let headerTotalTopMargin = Constants.BottomSheet.heroStackTopBottomMargin
+        + ( isExpandable
+            ? Constants.BottomSheet.headerExpandableAddedTopMargin
+            : Constants.BottomSheet.headerNonExpandableAddedTopMargin)
+        bottomSheetHeroStackTopConstraint?.constant = headerTotalTopMargin
+
+        let oldHeaderHeight = bottomSheetController.collapsedContentHeight
+        let newHeaderHeight = headerTotalTopMargin + maxHeroItemHeight + Constants.BottomSheet.heroStackTopBottomMargin
+
+        if newHeaderHeight != oldHeaderHeight {
```

Comment:
so the hero height changes based on one line or 2 lines? its not the same and the padding is bit different?

Reply:
Yeah, it does change and it's what makes this part of the spec so hard to implement. Visually, with the sheet expanded the header size almost doesn't change. But when collapsed we need to still show all of the tall labels without leaking out of the safe area, so the collapsed height has to change.
I'm working on a separate PR that simplifies the sizing and I'll update this PR afterwards to hopefully simplify this as well.

Reply:
Nevermind, I did all of it here.

So, your suggestion was great and it allowed me to make things more straightforward.
The header view height is now always the same. The 1 vs. 2 line hero label now just decides how much more of it we need to show when collapsed to fulfill the bottom whitespace requirement. 
The new solution also accounts for difference in bottom safe area insets. So if there is enough of an inset, we subtract that from how much more whitespace we need. If there isn't enough, we add our own.